datalad/customremotes/archives.py

```diff
@@ -424,18 +424,12 @@ def _annex_get_archive_by_key(self, akey):
             % (naturalsize(akey_size) if akey_size else "unknown")
         )
 
-        def progress_indicators(l):
-            self.info("PROGRESS-JSON: " + l.rstrip(os.linesep))
-
-        self.runner(["git", "annex", "get",
-                     "--json", "--json-progress",
-                     "--key", akey
-                     ],
-                    log_stdout=progress_indicators,
-                    log_stderr='offline',
-                    # False, # to avoid lock down
-                    log_online=True,
-                    cwd=self.path, expect_stderr=True)
+        # call git annex get without capturing any output
+        self.repo._git_runner.run(
+            ["git", "annex", "get", "--json", "--json-progress", "--key", akey],
+            protocol=None,
+            cwd=self.path,
+        )
```

Comment:
This change leads to a protocol error.

<details>
<summary>demo</summary>

```
#!/bin/sh

set -eu

cd "$(mktemp -d "${TMPDIR:-/tmp}"/dl-XXXXXXX)"
datalad create
datalad download-url --archive \
        https://github.com/datalad/datalad-ukbiobank/archive/0.3.tar.gz
git annex drop --all
datalad get datalad-ukbiobank-0.3/README.md
```

</details>

On base (8081fcf6f1):

```
[...]
[INFO] To obtain some keys we need to fetch an archive of size 51.5 kB 
[INFO] PROGRESS-JSON: {"command":"get","note":"from web...\nchecksum...","success":true,"input":[],"key":"MD5E-s51496--03f0fc3fa1ded1efbffbf0f0b435ef6f.tar.gz","file":null} 
get(ok): datalad-ukbiobank-0.3/README.md (file) [from datalad-archives...]
```

After this PR:

```
[...]
[INFO] To obtain some keys we need to fetch an archive of size 51.5 kB 
get(ok): datalad-ukbiobank-0.3/README.md (file) [external special remote protocol error, unexpectedly received "{"byte-progress":34861,"action":{"command":"get","note":"from web...","input":[],"key":"MD5E-s51496--03f0fc3fa1ded1efbffbf0f0b435ef6f.tar.gz","file":null},"total-size":51496,"percent-progress":"67.7%"}" (unable to parse command)
unable to use special remote due to protocol error]
```

Reply:
Arggh, true. Thanks for the demo! I have pushed a fix that adds a protocol to handle the JSON output.